core/domain/prod_validation_jobs_one_off.py

```diff
@@ -110,67 +351,3183 @@ def validate(cls, item):
+            if item.exploration_model_last_updated != last_human_update_time:
+                cls.errors['exploration model last updated check'].append((
+                    'Entity id %s: The exploration_model_last_updated '
+                    'field: %s does not match the last time a commit was '
+                    'made by a human contributor: %s') % (
+                        item.id, item.exploration_model_last_updated,
+                        last_human_update_time))
+
+    @classmethod
+    def _validate_related_model_properties(cls, item):
+        """Validate that model properties match the corresponding exploration
+        model and exploration rights model properties.
+
+        Args:
+            item: ExpSummaryModel to validate.
+        """
+        _, exploration_model_tuples_list = cls.external_models[
+            'exploration_ids']
+        _, exploration_rights_model_tuples_list = cls.external_models[
+            'exploration_rights_ids']
+
+        exploration_model_properties_dict = {
+            'title': 'title',
+            'category': 'category',
+            'objective': 'objective',
+            'language_code': 'language_code',
+            'tags': 'tags',
+            'exploration_model_created_on': 'created_on',
+        }
+
+        for (_, exploration_model) in exploration_model_tuples_list:
+            for (property_name, exploration_model_property_name) in (
+                    exploration_model_properties_dict.iteritems()):
+                value_in_summary_model = getattr(item, property_name)
+                value_in_exploration_model = getattr(
+                    exploration_model, exploration_model_property_name)
+
+                summary_model_output_value = value_in_summary_model
+                if isinstance(summary_model_output_value, list):
+                    summary_model_output_value = (',').join(
+                        summary_model_output_value)
+
+                exploration_model_output_value = (
+                    value_in_exploration_model)
+                if isinstance(exploration_model_output_value, list):
+                    exploration_model_output_value = (',').join(
+                        exploration_model_output_value)
+
+                if value_in_summary_model != value_in_exploration_model:
+                    cls.errors['%s field check' % property_name].append((
+                        'Entity id %s: %s field in entity: %s does not match '
+                        'corresponding exploration %s field: %s') % (
+                            item.id, property_name, summary_model_output_value,
+                            exploration_model_property_name,
+                            exploration_model_output_value))
+
+
+        exploration_rights_model_properties_dict = {
+            'first_published_msec': 'first_published_msec',
+            'status': 'status',
+            'community_owned': 'community_owned',
+        
```

Comment:
Here and elsewhere, please **never** copy/paste large chunks of code unless you have a very good reason to. Find ways to reuse code. See the [DRY principle](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself).

Reply:
Updated.

Reply:
Thanks for this. It makes the jobs file much cleaner and readable :)

Reply:
You're welcome, but I still see a lot of copy/paste. Why don't you just reuse _validate_topic_similarities() and other existing functions where possible? (You can make it public if you need to use it here.)

Reply:
Updated now.